aptos-move/move-examples/dao/nft_dao/sources/nft_dao.move

```diff
@@ -0,0 +1,623 @@
+    use dao_platform::token_type::get_token_type;
+    use std::bcs;
+    use std::error;
+    use std::signer;
+    use std::string::String;
+    use std::string;
+    use std::vector;
+
+    /// The new address to be delegated to already exists in the platform
+    const EDELEGATED_ADDRESS_ALREADY_EXISTS: u64 = 1;
+
+    /// This account doesn't have enough voting weights
+    const EVOTING_WEIGHTS_NOT_ENOUGH: u64 = 2;
+
+    /// This account doesn't own this DAO's voting token
+    const ENOT_OWN_THE_VOTING_DAO_TOKEN: u64 = 3;
+
+    /// This function is supported in proposal
+    const ENOT_SUPPROTED_FUNCTION: u64 = 4;
+
+    /// Can only propose a start time in future
+    const EPROPOSAL_START_TIME_SHOULD_IN_FUTURE: u64 = 5;
+
+    /// Cannot only use membership token of the DAO
+    const ENOT_FOUND_ADMIN_ACCOUNT: u64 = 6;
+
+    /// String length exceeds limits
+    const ESTRING_TOO_LONG: u64 = 7;
+
+    /// Proposal ended
+    const EPROPOSAL_ENDED: u64 = 8;
+
+    /// Propsoal already resolved
+    const EPROPOSAL_RESOLVED: u64 = 9;
+
+
+    /// TOKEN VOTED FOR PROPOSAL
+    const ETOKEN_ALREADY_VOTED: u64 = 10;
+
+    /// Resource Account Doesn't exist
+    const ERESOURCE_ACCT_NOT_EXIST: u64 = 11;
+
+    /// Proposal resource not created
+    const EPRPOSALS_NOT_EXIST_AT_ADDRESS: u64 = 12;
+
+    /// Proposal id doesn't exist
+    const EPRPOSAL_ID_NOT_EXIST: u64 = 13;
+
+
+    /// Constants
+    const PROPOSAL_UNRESOLVED: u8 = 0;
+    const PROPOSAL_RESOLVED_PASSED: u8 = 1;
+    const PROPOSAL_RESOLVED_NOT_PASSED: u8 = 2;
+
+    struct DAO has key {
+        /// Name of the DAO
+        name: String,
+        /// The threshold that the proposal can resolve, which is an absolute number of NFT voted
+        resolve_threshold: u64,
+        /// The NFT Collection that is used to govern the DAO
+        governance_token: GovernanceToken, // This is the governance token or NFT used for voting
+        /// The voting duration in secs
+        voting_duration: u64,
+        /// Minimum weight for proposal
+        min_required_proposer_voting_power: u64,
+        /// Proposal counter
+        next_proposal_id: u64,
+    }
+
+    /// The collection should only contains NFTs, where all token name only has 1 maximal and immutable
+    /// The total supply is fixed with the token names.
+    struct GovernanceToken has store {
+        /// The creator address of this NFT collection
+        creator: address,
+        /// The collection name
+        collection: String,
+    }
+
+    /// All proposals
+    struct Proposals has key {
+        proposals: Table<u64, Proposal>,
+    }
+
+    /// Store the general information about a proposal
+    struct Proposal has copy, drop, store {
+        /// Name of the proposal, limiting to 64 chars
+        name: String,
+        /// Description of the proposal, limiting to 512 chars
+        description: String,
+        /// The name of function to be executed
```

Comment:
If we only support specific actions, why not just use an enum (u64) instead of function name?

Reply:
easy to remember and use by caller